### PR TITLE
Switch to got

### PR DESCRIPTION
**Pull request by kg6zvp**

### CODE REVIEW

File: src/fetch/fetchURL.js [hunk 1]
```diff
@@ -17,22 +16,26 @@ const httpError = (err, res) => {
   const nerror = rewriteError(err)
   nerror.requestError = true
   nerror.code = res.code
-  nerror.status = res.status
+  nerror.status = res.statusCode
   nerror.headers = res.headers
-  nerror.body = res.text
```
Comment (yocontra) [inline src/fetch/fetchURL.js:22]: We still need to keep this behavior - the body should be included with the error. In the tests the body just happens to be the status code.

File: src/fetch/fetchURL.js [hunk 2]
```diff
@@ -53,7 +56,7 @@ export default (url, { headers, timeout }={}) => {
   })
   out.abort = () => {
     hardClose(out)
-    req.abort()
+    req.end()
```
Comment (yocontra) [inline src/fetch/fetchURL.js]: Does this cancel the outbound request or just end the stream?

**Review by yocontra — CHANGES_REQUESTED** [review]

File: src/fetch/fetchURL.js (hunk 1 above)
Comment (yocontra) [reply src/fetch/fetchURL.js:22]: You can update the test bodies to make sure this doesn't get overlooked again ^

File: src/fetch/fetchURL.js [hunk 3]
```diff
@@ -17,22 +16,26 @@ const httpError = (err, res) => {
...
-    .retry(10)
-  if (timeout) req = req.timeout(timeout)
-  if (headers) req = req.set(headers)
+  let gotOptions = {
+    stream: true,
+    buffer: false,
+    followRedirects: true,
+    retry: 10
+  }
+  if (timeout) gotOptions.timeout = timeout
```
Comment (yocontra) [inline src/fetch/fetchURL.js]: you can move these straight into the object

File: src/fetch/fetchURL.js [hunk 4]
```diff
@@ -17,22 +16,26 @@ const httpError = (err, res) => {
...
+  if (headers) gotOptions.headers = headers
+
+  let req = got(url, gotOptions)
   req
     // http errors
     .once('response', async (res) => {
```
Comment (yocontra) [inline src/fetch/fetchURL.js]: this block was made specific for superagent, is this still working?

Comment (kg6zvp) [reply src/fetch/fetchURL.js]: It is still working! I was surprised, but I think it's because they're both using request under the hood.

**Pull request merged**